### ISSUE TITLE
Execution Already Exists Issue

### ISSUE THREAD

**Issue by neehar-pylon** [CloudSnorkel/cdk-github-runners#607]
I am getting this error in the cloud watch when I try to run a self hosted runner one after the other and it is not able retrieve installation Id from the webhook, has anyone faced this issue before.

```
{
    "timestamp": "2024-08-15T22:26:51.006Z",
    "level": "ERROR",
    "requestId": "",
    "message": {
        "errorType": "ExecutionAlreadyExists",
        "errorMessage": "Execution Already Exists: 'secretsarn'",
        "stackTrace": [
            "ExecutionAlreadyExists: Execution Already Exists: 'secretsarn'",
            "    at de_ExecutionAlreadyExistsRes (/var/runtime/node_modules/@aws-sdk/client-sfn/dist-cjs/index.js:2125:21)",
            "    at de_CommandError (/var/runtime/node_modules/@aws-sdk/client-sfn/dist-cjs/index.js:2070:19)",
            "    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)",
            "    at async /var/runtime/node_modules/@aws-sdk/node_modules/@smithy/middleware-serde/dist-cjs/index.js:35:20",
            "    at async /var/runtime/node_modules/@aws-sdk/node_modules/@smithy/core/dist-cjs/index.js:165:18",
            "    at async /var/runtime/node_modules/@aws-sdk/node_modules/@smithy/middleware-retry/dist-cjs/index.js:320:38",
            "    at async /var/runtime/node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:34:22",
            "    at async Runtime.handler (/var/task/index.js:13379:21)"
        ],
        "name": "ExecutionAlreadyExists",
        "$fault": "client",
        "$metadata": {
            "httpStatusCode": 400,
            "requestId": "",
            "attempts": 1,
            "totalRetryDelay": 0
        },
        "__type": "com.amazonaws.swf.service.v2.model#ExecutionAlreadyExists"
    }
}
```

**Comment by kichik** [comment]
Where does `secretsarn` come from? What makes you say it wasn't able to retrieve the installation id from the webhook?

**Comment by neehar-pylon** [comment]
I followed every step mentioned in the readme.md, I setup a new repo app  and able to invoke the lambda function to get the `status.json` which has necessary information including the `installations` object and all the status are `ok`, I created a sample workflow file to see if I am able to run it. But when I try to raise a PR in the AWS step function it is failing and the reason is installation id is -1. 

`installationId: payload.installation?.id ?? -1,` I can see from the code that in library if the webhook payload does not contain the installation id then it by default assigns it to -1. 

In my case sometimes when I invoke the lambda  status function before raising a PR, I am able to see the installation id and everything working correctly but it does not work every time, also it should not be required to run the status lambda function every time before you raise a PR right ?

 I checked the GitHub app and permissions and everything and it is setup correctly, I was also not able to find the installation Id in the webhook payload from the webhook URL which was created while I created the app. 
 
 `secretsarn` come from the GitHub app.

**Comment by kichik** [comment]
You don't need to call the status function more than once while setting things up.

Installation id would not be there if the webhook is not called from an app. Did you maybe create an additional webhook manually? Was the initial setup done using the one-click wizard or did you create the app yourself?

Is `secretsarn` actually what's coming from the webhook or are you censoring something? That would be the actual cause of issue here. The execution id cannot be something constant as new executions will fail. It should never be anything like just `secretsarn`.

https://github.com/CloudSnorkel/cdk-github-runners/blob/ce649c33d9cac18e2dd81752db03d4f39d4f790d/src/webhook-handler.lambda.ts#L178-L181

**Comment by neehar-pylon** [comment]
No,  `secretsarn` is just an alias I used for censoring. Yes, I setup the app using the one-click wizard as mentioned in the `Readme.md`. 

Also there is one more thing on successful runs I was able to see the installation id from the recent webhook deliveries from the Github App, but when looked from the `Webhook` section of the GitHub repo settings there is no installation id present.

Also wanted to ask do I need to Add a Webhook manually in the `Webhooks` field in the repo settings ?
